### PR TITLE
adapters and service layer for Register and Login User and usecase CRUD operations

### PR DESCRIPTION
**Pull request by usmanov0**

### CODE REVIEW

File: internal/users/app/user_usecase.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+func NewUserUseCase(userRepo domain.UserRepository) UserUseCase {
+	return &userUseCase{userRepo: userRepo}
+}
+
+func (u userUseCase) SignUpUser(user *domain.NewUser) (int, error) {
+	userFactory := u.userFac.CreateNewUser(user)
+
+	err := validateUserInfoForSignUp(
```
Comment (khudayberdiyev25) [inline internal/users/app/user_usecase.go]: maybe u write such validations inside `userfactory.CreateNewUser` ?

File: internal/users/app/user_usecase.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	return id, nil
+}
+
+func (u userUseCase) SignInUser(email, password string) (bool, error) {
+	err := validateUserInfoForSignIn(email, password)
+
+	if err != nil {
+		return false, err
+	}
+	exists, _ := u.userRepo.ExistsByMail(email)
```
Comment (khudayberdiyev25) [inline internal/users/app/user_usecase.go:55]: why u don't validate passport ?

**Review by khudayberdiyev25 — CHANGES_REQUESTED** [review]

File: internal/users/app/user_usecase.go (hunk 2 above)
Comment (usmanov0) [reply internal/users/app/user_usecase.go:50]: we don't have field like passport

File: internal/users/domain/user_factory.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+import "time"
+
+type UserFactory struct{}
+
+func NewUserFactory() *UserFactory {
+	return &UserFactory{}
+}
+
+func (f UserFactory) CreateNewUser(user *NewUser) *User {
```
Comment (xbakhrom) [inline internal/users/domain/user_factory.go:16]: I think arguments must be username, email, password .

Comment (usmanov0) [reply internal/users/domain/user_factory.go:16]: appreciate it.

**Pull request merged**